### PR TITLE
ttcn: fix parsing string that contains double quotes escaped by backslash or double quotes  

### PR DESCRIPTION
**Pull request by akrzyz**
Fix bug that wrongly interprets strings with double quote.
TTCN has two ways to put double quote into charstring:
1. escape by backslash \\"
2. two double quotes ""

new unit ttcn-string-with-quotes is added

### CODE REVIEW

File: parsers/ttcn.c [hunk 1]
```diff
@@ -438,20 +438,33 @@ static ttcnToken_t * getToken (void)
 		/* Charstring */
 		pTtcnToken->type = T_LITERAL;
 		pTtcnToken->value = vStringNew();
+        vStringPut(pTtcnToken->value, c);
 		do
 		{
-			vStringPut(pTtcnToken->value, c);
 			c = getcFromInputFile();
+			vStringPut(pTtcnToken->value, c);
```
[the remark this answers is not in the document]
Comment (akrzyz) [reply parsers/ttcn.c]: I changed do-while to while and this c != EOF is checked now in while(..)

File: parsers/ttcn.c [hunk 2]
```diff
@@ -438,20 +438,33 @@ static ttcnToken_t * getToken (void)
...
+            /* consume escaped chracters */
+            if(c == '\\')
+            {
+                c = getcFromInputFile();
+                vStringPut(pTtcnToken->value, c);
```
[the remark this answers is not in the document]
Comment (akrzyz) [reply parsers/ttcn.c]: this one is checked now in if above

File: parsers/ttcn.c [hunk 3]
```diff
@@ -438,20 +438,31 @@ static ttcnToken_t * getToken (void)
...
-					continue;
-				ungetcToInputFile(c2);
+                /* consume "" */
+				if (c2 == '"')
+                {
+                    vStringPut(pTtcnToken->value, c2);
+                    continue;
+                }
+                /* c is " that close string, c2 is out of string */
+                ungetcToInputFile(c2);
```
Comment (masatake) [inline parsers/ttcn.c]: c2 can hold EOF here.
Passing EOF to ungetcToInputFile is legal.
However, I think "if (c2 != EOF) ungetcToInputFile(c2);" is better.

**Comment by masatake** [comment]
Thank you. Squashed when merging.

**Pull request merged**